sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.http.rest.PagedFlux;
+import java.util.List;
+import java.util.Map;
+import reactor.core.publisher.Mono;
+
+/**
+ * class for the table async client
+ */
+@ServiceClient(
+    builder = TableClientBuilder.class,
+    isAsync = true)
+public class TableAsyncClient {
+    private final String tableName;
+
+    TableAsyncClient(String tableName) {
+        this.tableName = tableName;
+    }
+
+    /**
+     * Queries and returns entities in the given table using the select and filter strings
+     *
+     * @param queryOptions the odata query object
+     * @return a paged flux of all the entity which fit this criteria
+     */
+    public PagedFlux<TableEntity> queryEntity(QueryOptions queryOptions) {
+        return null;
+    }
+
+    /**
+     * Queries and returns entities in the given table with the given rowKey and ParitionKey
+     *
+     * @param rowKey the given row key
+     * @param partitionKey the given partition key
+     * @return a list of the tables that fit the row and partition key
+     */
+    public PagedFlux<TableEntity> queryEntitiesWithPartitionAndRowKey(String rowKey, String partitionKey) {
+        return null;
+    }
+
+    /**
+     * insert a TableEntity with the given properties and return that TableEntity. Property map must include
+     * rowKey and partitionKey
+     *
+     * @param tableEntityProperties a map of properties for the TableEntity
+     * @return the created TableEntity
+     */
+    public Mono<TableEntity> createEntity(Map<String, Object> tableEntityProperties) {
```

Comment:
We should also have `*EntityWithResponse()` methods for all APIs that return `Mono<>` but if the intent of this PR is to just show the code snippets, you can ignore this comment.